tests/test_config.cpp

```diff
@@ -186,3 +186,42 @@ SCENARIO("Check Config Constructor", "[!throws]") {
         }
     }
 }
+
+SCENARIO("Check everest config parsing", "[!throws]") {
+    auto bin_dir = Everest::tests::get_bin_dir().string() + "/";
+    GIVEN("A complete and patched valid config") {
+        auto config = Everest::load_yaml(bin_dir + "valid_complete_config.json");
+        THEN("It should not throw") {
+            CHECK_NOTHROW(everest::config::parse_everest_config(config));
+        }
+    }
+    GIVEN("A valid config that misses a setting") {
```

Comment:
@hikinggrass : At the point we parse the config, should we make sure that all settings are present (fallback to defaults)? In the `Settings` type all parameters are required (not std::optional)